username_api.py

```diff
@@ -16,15 +17,37 @@ def check_username(website, username):
 	'behance'   :'https://{}.net/{}'.format(website, username)
 	}.get(website, 'https://{}.com/{}'.format(website, username)) # default
 
-	if website in ['pinterest', 'gitlab']:
-		res  = r.get(url)
-		code = 200 if bytes(username, encoding='utf-8') in res.content \
-			else 404
+	if check_format(website, username):
+		if website in ['pinterest', 'gitlab']:
+			res  = r.get(url)
+			code = 200 if bytes(username, encoding='utf-8') in res.content \
+				else 404
 
-		return {'status': code, 'url': url}
-	
+			return {'status': code, 'url': url}
+
+		else:
+			return {'status': r.get(url).status_code, 'url': url}
+	else:
+		return {'status': 400, 'url': url}
```

Comment:
this is not a good use of `400`, as it conflicts with actual `400`, which is about the http request , not the application logic.

The client made a correctly framed request, which was able to be correctly understood, and the app can respond sanely to inform the client that their request is invalid due to rules which the client doesnt know about.

Reply:
Agreed, makes sense.